### PR TITLE
accurately fetching main UP! controller

### PR DESCRIPTION
**Pull request by tehnalogos**

### CODE REVIEW

File: frontend/components/wallet/WalletProvider.tsx [hunk 1]
```diff
@@ -53,32 +54,70 @@ export const WalletProvider: React.FC<Props> = ({ children }) => {
...
         // Update state and localStorage with the first account address.
         setAccount(accounts[0]);
+        // To enable the Sign-In With Ethereum (SIWE) screen, you need to prepare a message with a specific format
+        const siweMessage = new SiweMessage({
+          domain: window.location.host, // Domain requesting the signing
+          address: accounts[0], // Address performing the signing
+          statement: 'By logging in you agree to the terms and conditions.', // a human-readable assertion user signs
+          uri: window.location.origin, // URI from the resource that is the subject of the signing
+          version: '1', // Current version of the SIWE Message
+          chainId: 4201, // Chain ID to which the session is bound, 4201 is LUKSO Testnet
```
Comment (democrazyfi) [inline frontend/components/wallet/WalletProvider.tsx:93]: should this be in constants so we can later parameterise it for mainnet?

File: frontend/components/wallet/WalletProvider.tsx [hunk 2]
```diff
@@ -53,32 +54,70 @@ export const WalletProvider: React.FC<Props> = ({ children }) => {
...
+        if (accounts.length) {
+          disconnect();
+        }
         const message =
           error && error.error && error.error.message
             ? error.error.message
             : 'An unknown error occurred';
-        console.log(`Connection error: ${message}`);
+        const toastMessage = `Connection error: ${message}`;
+        console.log(toastMessage);
```
Comment (democrazyfi) [inline frontend/components/wallet/WalletProvider.tsx:118]: we should probably `console.error(message)` so we get the full trace here

File: frontend/components/wallet/WalletProvider.tsx [hunk 3]
```diff
@@ -53,32 +54,70 @@ export const WalletProvider: React.FC<Props> = ({ children }) => {
...
+      let accounts: string[] = [];
       try {
         // Reset the graveVault address when connecting
         setGraveVault(undefined);
         // Request accounts from the wallet.
-        const accounts = await web3.eth.requestAccounts();
+        accounts = await web3.eth.requestAccounts();
         console.log('Connected with', accounts[0]);
         // Update state and localStorage with the first account address.
         setAccount(accounts[0]);
```
Comment (democrazyfi) [inline frontend/components/wallet/WalletProvider.tsx:85]: I know we didn't handle it previously but perhaps something we can do in another PR is check if accounts array is empty? I guess that's a scenario where there is an extension but not accounts setup yet

File: frontend/constants/networks.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+export const NETWORKS = {
```
Comment (tehnalogos) [inline frontend/constants/networks.ts:1]: realize this slightly differs from our current "constants" setup, but im just getting processes started for testnet vs mainnet distinction and tis is a better structure. future PR dealing with it should be responsible for full clean up rather than doing it in this PR

**Review by Ledespana — APPROVED** [review]
